From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Adams <chris.adams@jollamobile.com>
Date: Thu, 27 Mar 2014 18:58:03 +1000
Subject: [PATCH] OAuth2: Relax RefreshToken restriction on ProvidedTokens

Some OAuth2 services (like Facebook) do not return RefreshToken
data, only AccessToken and ExpiresIn.  This commit ensures that
clients can still provide cached token data when no RefreshToken
is present in the token map.
---
 src/oauth2plugin.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/oauth2plugin.cpp b/src/oauth2plugin.cpp
index 57df6d8819d5a14e8eb2090728f9dc023cd7f743..be39263083894ae7e190514f92c4996421eebba2 100644
--- a/src/oauth2plugin.cpp
+++ b/src/oauth2plugin.cpp
@@ -336,9 +336,8 @@ void OAuth2Plugin::process(const SignOn::SessionData &inData,
         OAuth2PluginTokenData providedTokens =
                 SignOn::SessionData(tokens.ProvidedTokens())
                 .data<OAuth2PluginTokenData>();
-        if (providedTokens.AccessToken().isEmpty() ||
-            providedTokens.RefreshToken().isEmpty()) {
-            //note: we don't check ExpiresIn as it might not be required
+        if (providedTokens.AccessToken().isEmpty()) {
+            //note: we don't check RefreshToken or ExpiresIn as it might not be required
             TRACE() << "Invalid provided tokens data - continuing normal process flow";
         } else {
             TRACE() << "Storing provided tokens";
